### ISSUE TITLE
ArToolkit out of date?

### ISSUE THREAD

**Issue by santosluiz** [artoolkit/arunity5#164]
Always when I start a project, the console returns several troubleshoots in ArToolkit Unity Package.

Example:

Always when I import the ArToolkit package for unity, several erros about obsolete code and genral troubles are displayed. Sometimes I have to comment the line(s) with error(s).

[Like this](http://imgur.com/7cV05dB) and [and this](http://imgur.com/yygpEoI)

(Maybe it may be incompatibility with new versions of unity3d)

Well, in general way, I would like to know about the updates of project or some relevant/interesting news that will help the ArToolkit community about the update of project.

Best regards,
Luiz

**Comment by ThorstenBux** [comment]
Both of these errors are well known but shouldn't trouble you when you work with ARToolKit5 in a Unity3D project. The warnings are due to some legacy code inside the ARToolKit5 plugin but that still work fine and will be resolved once ARToolKit6 is released. The error is a nasty thing regarding the package and how Unity evaluates it. But as said it is annoying that is shows these errors but they shouldn't keep you from using ARToolKit5 in your projects. Let me know if you experience otherwise.

**Comment by santosluiz** [comment]
For solve this error I just have to comment the error line.

About ARToolKit6, is there an average of when it will be release? It will be support for HoloLens?

**Comment by tsteffelbauer** [comment]
According to this post, native support for Hololens won't be a thing too soon and the release of ARToolkit6 will take a while, too
https://artoolkit.org/community/forums/viewtopic.php?f=24&t=28105

**Comment by ThorstenBux** [comment]
@tsteffelbauer I wouldn't say that the ARToolKit6 release will be too far off. In fact, we are preparing for a public beta release right now.

**Comment by RakeshKoplodR** [comment]
TypeLoadException: Could not load type 'UnityEngine.Cloud.Service.CloudServiceConfig' from assembly 'UnityEngine, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null'.
System.Reflection.MonoMethodInfo.GetMethodInfo (IntPtr handle) (at /Users/builduser/buildslave/mono/build/mcs/class/corlib/System.Reflection/MonoMethod.cs:59)
System.Reflection.MonoMethodInfo.GetAttributes (IntPtr handle) (at /Users/builduser/buildslave/mono/build/mcs/class/corlib/System.Reflection/MonoMethod.cs:75)
System.Reflection.MonoMethod.get_Attributes () (at /Users/builduser/buildslave/mono/build/mcs/class/corlib/System.Reflection/MonoMethod.cs:245)
System.Reflection.MethodBase.get_IsSpecialName () (at /Users/builduser/buildslave/mono/build/mcs/class/corlib/System.Reflection/MethodBase.cs:184)
UnityEditor.Build.BuildPipelineInterfaces.InitializeBuildCallbacks (Boolean findBuildProcessors, Boolean findSceneProcessors) (at /Users/builduser/buildslave/unity/build/Editor/Mono/BuildPipeline/BuildPipelineInterfaces.cs:149)
UnityEditor.BuildPlayerWindow:BuildPlayerAndRun()

**Comment by ThorstenBux** [comment]
Hi @RakeshKoplodR , I need more info here. What Unity version are you using? Can you make a screen shot so I can see the errors in the console of Unity? Thanks

**Comment by ThorstenBux** [comment]
Just updated the package again maybe that helps

**Comment by RakeshKoplodR** [comment]
Hi @ThorstenBux  Thak you for quick response. I am using Unity 5.6.1f1.
Please find the attachment for the error screenshot.
<img width="1676" alt="screen shot 2017-06-06 at 10 07 45 am" src="https://cloud.githubusercontent.com/assets/12967064/26814129/b25c2520-4aa0-11e7-8d39-23add77fe5e5.png">

**Comment by ThorstenBux** [comment]
When you import the .package from gitHub make sure to only tick the components shown in the following screenshot 

<img width="808" alt="screen shot 2017-06-06 at 4 58 03 pm" src="https://cloud.githubusercontent.com/assets/16772796/26814401/7d36be58-4ad9-11e7-9582-7f45398bd31e.png">

**Comment by RakeshKoplodR** [comment]
Hi @ThorstenBux . I will try this. I want to share the screenshot of error what I am getting when I try to build the app. Sorry for inconvenience
<img width="1675" alt="screen shot 2017-06-06 at 10 28 33 am" src="https://cloud.githubusercontent.com/assets/12967064/26814515/d1cd6fa6-4aa3-11e7-8e14-7941e354bca7.png">
.

**Comment by ThorstenBux** [comment]
are you using any other plugin? This error doesn't look like an ARToolKit related error

**Comment by RakeshKoplodR** [comment]
No. Before importing ArToolkit package app is running perfectly without any errors. I might be doing something wrong. Below are the package components I am importing. I am getting same error again as above.

<img width="1239" alt="screen shot 2017-06-06 at 10 41 21 am" src="https://cloud.githubusercontent.com/assets/12967064/26814734/697d82c2-4aa5-11e7-87ad-574c835aa5f3.png">

**Comment by ThorstenBux** [comment]
When is your error happening exactly? When importing the package or when try to build to a specific platform? Which platform are you targeting?

**Comment by RakeshKoplodR** [comment]
The error is happening when trying to build app after importing the package. I am targeting for iOS (iPhone).

App was running before importing ARToolKit.

I am totally stuck here. Please help me.

**Comment by ThorstenBux** [comment]
Just to make sure. Can you import the package successful? 
Have you imported to package into an empty project and could you run one of the example scenes successfully? Please verify and tell me once you did this.

**Comment by RakeshKoplodR** [comment]
Hi @ThorstenBux  Thank you for your support. I have solved this issue. I have reinstalled Unity and imported package and it has imported without any error. 

Now I am facing new issue. On iPhone camera is not showing only black screen is shown on screen. I am following below steps:

https://artoolkit.org/documentation/doku.php?id=6_Unity:unity_getting_started

It will be great if you can help me with this.

**Comment by annapoth** [comment]
Same issue on my side. I dont want to reinstall Unity because I did this several times and still not working. Before that I created an project using the Vurforia SDK. Afterwards I created an empty project and imported the ARUnity5-5.3.2 package. I get the same error messages. 

Any hints would be very helpful!

**Comment by ThorstenBux** [comment]
Honestly, I'm a bit lost here myself. But maybe you would like to try ARToolKit6. You can get it from artoolkit.org website. It has a Unity plugin as well

**Comment by annapoth** [comment]
I tried that already but Unity crashed completely when importing ARToolkit6 on MacOsX

**Comment by ThorstenBux** [comment]
Wow, never had that. Any crash report that I could look at?

**Comment by annapoth** [comment]
Ah now it was working. But there are so many errors they also appear if I create an empty project? 
So I can't get the demo scenes running. 
<img width="1440" alt="bildschirmfoto 2017-06-20 um 11 17 44" src="https://user-images.githubusercontent.com/7071632/27325930-45c5698e-55aa-11e7-85ed-353f466f64e8.png">

**Comment by ThorstenBux** [comment]
Ahhhh this ugly UNetWeaver errors. I have no idea where they come from. We had those with older versions of unity but I'm running the same version you do and never saw them. So it is hard to reproduce that.

**Comment by annapoth** [comment]
And this TypeLoadException: Could not load type 'UnityEngine.Cloud.Service.CloudServiceConfig' from assembly 'UnityEngine, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null' error? Because the UNetWeaver errors disappeared now..

**Comment by ThorstenBux** [comment]
The CloudService should not be part of the package of ARToolKit6. Maybe you can unselect the checkbox during import if it is. Otherwise you must have installed it to your Unity version and it seems not to be compatible

**Comment by annapoth** [comment]
Is there any way to uninstall it? Cant see the checkbox in the package import...

**Comment by ThorstenBux** [comment]
Try to only import this: 
<img width="805" alt="screen shot 2017-06-26 at 10 36 47 am" src="https://user-images.githubusercontent.com/16772796/27520280-6fb56f7c-5a5b-11e7-9728-c72ae8a7031d.png">
